lib/reverse_words.rb

```diff
@@ -1,6 +1,50 @@
 # A method to reverse each word in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n^2) where n is the length of my_words?
+# Space complexity: O(n) where n is the length of my my_words?
 def reverse_words(my_words)
-  raise NotImplementedError
+  return my_words if my_words == nil
+  return my_words if my_words.length == 1 || my_words.length == 0 || (my_words.length == 1 && my_words[0] == " ")
+  reverse = ""
+  new_string = ""
+  i = 0
+  while i < my_words.length
+    new_string += my_words[i]
+    if my_words[i] == " "
+      j = 0
+      k = new_string.length - 2
```

Comment:
Where possible, try to use more descriptive variable names. Too many singular letter variables makes it a lot harder to hold all of their meanings in my head.